### ISSUE TITLE
Accesing metatadata of current study

### ISSUE THREAD

**Issue by rcillavicomtech** [OHIF/Viewers#1475]
Hi folks,
We are developing a custom image annotation extension and we are interested in accessing Dicom metadata for the current visualized study. However, we have not found a straightforward way to do it while programming custom commands, not even StudyIDs or SeriesIDs. We have been poking around into the code and have found some data structures to host Dicom Metadata, but it appears to not be stored in neither the store or the sessionStorage. 
Is there a trivial way to access metadata for the current study without querying the Dicom server?
Kind Regards,
Rodrigo

**Comment by dannyrb** [comment]
Hello Rodrigo :wave: 

Always great to hear from people using extensions. I don't love the way we currently surface metadata:

- Add `@ohif/core` as a `peerDependency` for your extension
    - This will be provided by the `@ohif/viewer` project
- Do something like:

```js
import { utils } from '@ohif/core';

const { studyMetadataManager } = utils;
const studies = studyMetadataManager.all();

// viewportSpecificData and activeViewportIndex are exposed in redux under `viewports`
const { displaySetInstanceUid, studyInstanceUid } = viewportSpecificData[activeViewportIndex];

const study = studies.find(
    study => study.studyInstanceUID === studyInstanceUid
);
```

I know @JamesAPetts has a good number of changes near completion that clean up our metadata usage. Long term, we'll also want to abstract away the exposed state behind a user friendly interface. We would potentially inject something _like_ the studyMetadataManger to extensions so they don't need to add `@ohif/core` as a peerDependency.

Open to suggestions and PRs to improve this. We don't get as much real-world feedback as I would like from extension authors.

**Comment by NicholasDascanio** [comment]
Hi @dannyrb !
Thanks for the suggestion. I'm interested in accessing metadata of the current study too. I tried your solution it but I haven't been lucky. In particular, I'm not able to retrieve viewportSpecificData and activeViewportIndex.
I have imported redux and then I've used Object.getOwnPropertyNames() to get all properties that belongs to it. 
In fact, I have seen that  redux exposes "reducers", which in turn exposes "viewports", which, unfortunately, does not expose viewportSpecificData and activeViewportIndex. Am I missing something?

Kind regards,
Nicholas
